### ISSUE TITLE
How the batch graph changes its shape after GNN？

### ISSUE THREAD

**Issue by pancccool** [dmlc/dgl#3611]
There are 120 nodes in my graph, and each node has a 120-dimensional feature vector. Set batch size=20. After using the collate function and dataloader, the nodes of each batch graph are 2400 (20*120). After a layer of GNN, the shape becomes (2400, 60).
How do I change the output shape of GNN to (20, 120, 60), that is (batch, number of nodes, number of output features)？

Will the data be messed up when using .view（batch, batch_nodes, out_dim）?

**Comment by Rhett-Ying** [comment]
could you elaborate more on how you use/generate the batched graph? do you use NodeDataLoader with sampler and obtains `blocks` or call `dgl.batch()` on multiple graphs(each has 120 nodes)?

**Comment by pancccool** [comment]
> could you elaborate more on how you use/generate the batched graph? do you use NodeDataLoader with sampler and obtains `blocks` or call `dgl.batch()` on multiple graphs(each has 120 nodes)?


Yes, I use dgl.batch() to get the batched graph!
Because each of my graphs has 120 nodes, when I use the data_loader_direct, I will get the batched graph of 2400 nodes.
When the batched graph passes through the GNN layer, I hope to reshape the output to (batch, each_graph_nodes, out_dim).
What should I do？

            def collate(samples): 
                _graphs, labels = map(list, zip(*samples))
                batched_graph = dgl.batch(_graphs)
                return batched_graph, torch.tensor(labels)

            data_loader_direct = DataLoader(train_samples_direct, batch_size=20,
                                     collate_fn=collate)

**Comment by pancccool** [comment]
Thank you very much for your reply！

**Comment by pancccool** [comment]
Thank you very much for your reply!
I did not use the graph to get blocks

**Comment by Rhett-Ying** [comment]
graphs are batched sequentially, so you could safely reshape (2400,60) back to (20,120,60): `torch.reshape(x,(20,120,60)`

**Comment by pancccool** [comment]
> 图按顺序批处理，因此您可以安全地将 (2400,60) 整形回 (20,120,60)： `torch.reshape(x,(20,120,60)`

I understand, thank you very much！
